interfaces/cython/cantera/onedim.pyx

```diff
@@ -894,16 +904,15 @@ cdef class Sim1D:
 
         # Do initial steps without Soret diffusion
         soret_doms = [dom for dom in self.domains if getattr(dom, 'soret_enabled', False)]
-        for dom in soret_doms:
-            dom.soret_enabled = False
+        solve_soret = len(soret_doms)>0
```

Comment:
You could define `set_soret` after this point, and then not need to pass `soret_doms` in as an argument.

Reply:
done